### PR TITLE
build: contrib Dockerfile

### PR DESCRIPTION
**Pull request by djkazic**
Today we don't have a Dockerfile, this PR defines one.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.20-alpine
+
+WORKDIR /app
+RUN apk add --no-cache git
+RUN git clone https://github.com/lightninglabs/lightning-faucet.git .
+RUN go mod download
```
Comment (Roasbeef) [inline Dockerfile]: Is this actually necessary? The `go build` commands should take care of resolving the deps.

Comment (Roasbeef) [reply Dockerfile]: From the docs: 
> The go mod download command is useful mainly for pre-filling the module cache or for loading data to be served by a [module proxy](https://go.dev/ref/mod#glos-module-proxy).

https://go.dev/ref/mod#go-mod-download

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.20-alpine
+
+WORKDIR /app
+RUN apk add --no-cache git
+RUN git clone https://github.com/lightninglabs/lightning-faucet.git .
```
Comment (Roasbeef) [inline Dockerfile:5]: Not necessarily blocking, but we should likely start to tag versions of the project. I can push one out at the current commit once this is merged.

File: Dockerfile (hunk 1 above)
Comment (djkazic) [reply Dockerfile]: Will remove

**Comment by djkazic** [comment]
Ready for re-review

**Review by Roasbeef — APPROVED** [review]
LGTM 🥦

**Pull request merged**